chainstate/src/config.rs

```diff
@@ -28,6 +28,7 @@ make_config_setting!(
         DEFAULT_MAX_IMPORT_BUFFER_SIZE,
     )
 );
+make_config_setting!(TxIndexEnabled, bool, true);
```

Comment:
While thinking about this, I think there's a design mistake here.

First, the default should be false. Because most users will use their wallets to index. However, this configuration is kind of invasive, because once enabled, people will never be able to run the software, ever again, without specifying the flag. I think this should have 3 different types:

- Enabled
- Disabled
- Unspecified

This way, users can set this only once on initialization, then forget about it.

While this sounds nice, it has the ugly side effect that we'll again have the problem of having to put the default of unspeficied somewhere. Maybe we should update `make_config_setting!` to support such cases. I'm not sure.

Reply:
I changed default to false.

Maybe we could leave it as-is for now (without switching to tri-state)? Because if we implement re-indexing someday this won't be a problem.